artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/LinkedListImpl.java

```diff
@@ -72,9 +115,43 @@ public void addHead(E e) {
          node.next.prev = node;
       }
 
+      itemAdded(node, e);
+
       size++;
    }
 
+   @Override
+   public E removeWithID(long id) {
+      if (nodeMap == null) {
+         return null;
+      }
+
+      Node<E> node = nodeMap.get(id);
+      if (node == null) {
+         return null;
+      }
+
+      if (node.prev != null) {
+         removeAfter(node.prev);
+      }
```

Comment:
if node != null, not.prev will always be != null...

We are adding the nodes to a separate list... the head will also be on the nodeList.


the != null is actually my OCD being extra careful. if you are strong opinioned about it I can take it out.

Reply:
Its more confusion and concern that it has modified the 'side area' map at this point but could return without removing anything from the the actual main list structure. If this is something to be careful about it seems more like it should be throwing an IllegalStateException or somthing rather than silently skipping, since the list must be broken for the check to have any effect.

Reply:
I removed the check..